executor/executor.go

```diff
@@ -78,12 +79,19 @@ func execCommand(path string, args []string) (err error) {
 			lgr.Debug.Printf("failed to read Stdin: %v", err)
 			return err
 		}
-		_, err = io.WriteString(stdin, string(b))
-		if err != nil {
-			lgr.Debug.Printf("failed to write StdinPipe: %v", err)
-			return err
-		}
-		stdin.Close()
+		go func() {
+			defer stdin.Close()
+			// Note: we must use goroutine,
+			// because when writing data exceeding pipe capacity this line is blocked until reading it.
+			_, err = io.WriteString(stdin, string(b))
```

Comment:
Is it better to use io.Copy()?  Then you do not need call ioutil.ReadAll()

Reply:
`io.Copy()`'s buffered reading and writing is very good. Thank you for your advice.

Reply:
I fixed it.